### PR TITLE
fixed: ZeroDivisionError: division by zero

### PR DESCRIPTION
**Pull request by sreio**
fixed: ZeroDivisionError: division by zero

for: https://github.com/deep-floyd/IF/issues/104#issuecomment-1542613470

### CODE REVIEW

**Comment by muellerzr** [comment]
@sreio can you do `pip install -e .[quality]` and run `make style; make quality`?

That will fix the failing check

**Review by sgugger — APPROVED** [review]
Thanks for the fix! Just one small comment.

File: src/accelerate/utils/modeling.py [hunk 1]
```diff
@@ -533,7 +533,7 @@ def get_balanced_memory(
     module_sizes = {n: v for n, v in module_sizes.items() if n not in leaves}
     # Once removed, leaves are the final modules.
     leaves = [n for n in module_sizes if len([p for p in module_sizes if n == "" or p.startswith(n + ".")]) == 0]
-    mean_leaves = int(sum([module_sizes[n] for n in leaves]) / len(leaves))
+    mean_leaves = int(sum([module_sizes[n] for n in leaves]) / ( len(leaves) or 1) )
```
[the remark this answers is not in the document]
Comment (sreio) [reply src/accelerate/utils/modeling.py:536]: Thanks for the reminder, I'll follow @muellerzr https://github.com/huggingface/accelerate/pull/1436#issuecomment-1549446098 and implement it

[the remark this answers is not in the document]
Comment (sreio) [reply src/accelerate/utils/modeling.py]: The max function should not be used because here it is necessary to prevent the divisor from being zero

**Comment by sreio** [comment]
@muellerzr I have fixed the grammar problem, Please merge the requests again, thanks!

File: src/accelerate/utils/modeling.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (sreio) [reply src/accelerate/utils/modeling.py]: code optimization

[the remark this answers is not in the document]
Comment (sreio) [reply src/accelerate/utils/modeling.py]: I'm sorry, I don't know how to merge your changes, I'll have to modify the commit code locally

[the remark this answers is not in the document]
Comment (sgugger) [reply src/accelerate/utils/modeling.py]: You can just click on the button Commit suggestion for next time :-)

**Review by sgugger — APPROVED** [review]

**Pull request merged**